### PR TITLE
[All] Replace last use of Qwt by QtCharts and remove internal library

### LINKED ISSUE #1496: [SOFAGUI_ENABLE_QWT] SOFA doesn't compile after the activation of the option  SOFAGUI_ENABLE_QWT
When you activated the option SOFAGUI_ENABLE_QWT in Sofa, you get compilation problem. This is maybe, due to the fact that QWT_VERSION_STR ==  "6.1.4" .
Adding line ```#include <QPainterPath>``` in the file ```extlibs/qwt-6.1.4/src/qwt_global.h``` makes SOFA compile. But still can't draw.  
______________________________________________________

Suggested labels:

### PR DESCRIPTION
**Pull request by epernod**
This PR include PR #1508 and replace last use of Qwt in QGraphDataWidget by qtCharts
Remove all cmake config of Qwt and remove the internal library

should fix #1496 or make things worst!

graph_cg
______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: modules/SofaGuiQt/CMakeLists.txt [hunk 1]
```diff
@@ -48,18 +48,6 @@ endif()
...
-option(SOFAGUI_ENABLE_QWT "Enable QWT for Qt GUIs")
-sofa_set_01(SOFAGUIQT_HAVE_QWT VALUE ${SOFAGUI_ENABLE_QWT} BOTH_SCOPES) # see SofaGuiQt.h.in
-if(SOFAGUI_ENABLE_QWT)
-    message("Qwt activated")
-    find_package(Qwt QUIET)
-    if(NOT QWT_FOUND)
-        add_subdirectory(${CMAKE_SOURCE_DIR}/extlibs/qwt-6.1.4/src ${CMAKE_BINARY_DIR}/extlibs/qwt)
-    endif()
-endif()
-
```
Comment (guparan) [inline modules/SofaGuiQt/CMakeLists.txt:62]: Do not forget to also propagate this removal in SofaGuiConfig.cmake.in :+1:

**Review by guparan — CHANGES_REQUESTED** [review]

File: modules/SofaGuiQt/src/sofa/gui/qt/config.h.in [hunk 2]
```diff
@@ -29,9 +29,7 @@
 #cmakedefine01 SOFAGUIQT_HAVE_QT5_CHARTS
 #cmakedefine01 SOFAGUIQT_HAVE_QT5_WEBENGINE 
 #cmakedefine01 SOFAGUIQT_HAVE_NODEEDITOR
-#cmakedefine01 SOFAGUIQT_HAVE_QWT
-
-#define SOFAGUI_ENABLE_NATIVE_MENU ${SOFAGUIQT_HAVE_QWT}
+#define SOFAGUI_ENABLE_NATIVE_MENU ${SOFAGUI_USE_NATIVE_MENU}
```
Comment (guparan) [inline modules/SofaGuiQt/src/sofa/gui/qt/config.h.in]: #cmakedefine01 SOFAGUI_ENABLE_NATIVE_MENU

File: modules/SofaGuiQt/CMakeLists.txt [hunk 3]
```diff
@@ -72,6 +60,8 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set(DEFAULT_SOFAGUI_ENABLE_NATIVE_MENU OFF)
 endif()
 option(SOFAGUI_ENABLE_NATIVE_MENU "Enable Qt to use the operating system's native menu for the Qt GUI" ${DEFAULT_SOFAGUI_ENABLE_NATIVE_MENU})
+sofa_set_01(SOFAGUI_USE_NATIVE_MENU VALUE ${SOFAGUI_ENABLE_NATIVE_MENU} BOTH_SCOPES)
+
```
Comment (guparan) [inline modules/SofaGuiQt/CMakeLists.txt]: Unnecessary

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by damienmarchal** [comment]
Hi guys... @guparan ask for change...but the PR is flagged as ready.

**Comment by guparan** [comment]
Changes have been done during the meeting, we're good to go :+1:

**Comment by damienmarchal** [comment]
Thanks @epernod

**Pull request merged**